### PR TITLE
Exercise that our rollup example works with ts_library

### PR DESCRIPTION
**Pull request by mrmeku**

### CODE REVIEW

File: examples/rollup/BUILD.bazel [hunk 1]
```diff
@@ -22,7 +28,7 @@ load(":rollup.bzl", "rollup")
 rollup(
     name = "bundle",
     entry_point = "examples/rollup/foo.js",
-    srcs = ["foo.js", "bar.js"],
+    deps = ["foo.js", "bar.js", ":mock_ts_lib"],
```
Comment (alexeagle) [inline examples/rollup/BUILD.bazel]: foo.js should still be in srcs, it's not the output of some other rule

File: examples/rollup/rollup.bzl [hunk 2]
```diff
@@ -11,17 +11,26 @@
...
 def _rollup(ctx):
   """This is not a full-featured Rollup bazel rule, just enough to test the
      other rules in this repo.
   """
-  args = ["--input", ctx.attr.entry_point]
+  inputs = collect_es6_sources(ctx)
+
+  entry_point = None
+  for i in inputs:
+    if ctx.attr.entry_point in i.path:
```
Comment (alexeagle) [inline examples/rollup/rollup.bzl]: what happens if we never set entry_point?

File: internal/collect_es6_sources.bzl [hunk 3]
```diff
@@ -29,15 +29,24 @@ def collect_es6_sources(ctx):
...
+          ("%s is neither a TypeScript nor a Closure JS library producing rule." % dep.label) +
+          "\nDependencies must be ts_library, ts_declaration, or closure_js_library.") 
 
-  rerooted_files = []
   for file in non_rerooted_files.to_list():
     rerooted_file = ctx.actions.declare_file(
-      "%s.es6/%s" % (
+      "%s.es6/node_modules/%s/%s" % (
         ctx.label.name,
+        ctx.workspace_name,
```
Comment (alexeagle) [inline internal/collect_es6_sources.bzl]: shouldn't it be ctx.label.workspace_root? what if the file comes from an external workspace?

File: internal/typescript_mock.bzl [hunk 4]
```diff
@@ -11,4 +11,22 @@ mock_typescript_lib = rule(
...
     "srcs": attr.label_list(allow_files = True),
   }
-)
\ No newline at end of file
+)
+
+def _mock_es6_typescript_lib(ctx):
+  transitive_es6_sources = []
+  for s in ctx.attr.srcs:
+    transitive_es6_sources += s.files.to_list()
```
Comment (alexeagle) [inline internal/typescript_mock.bzl]: `to_list` and back to depset is a smell, try

```
transitive_es6_sources = depset()
for s in ctx.attr.srcs:
  transitive_es6_sources = depset(transitive=[s.files, transitive_es6_sources])
```

File: internal/typescript_mock.bzl [hunk 5]
```diff
@@ -11,4 +11,22 @@ mock_typescript_lib = rule(
...
+  return struct(typescript = struct(
+    es5_sources = depset(),
+    transitive_es6_sources = depset(transitive_es6_sources)
+  ))
+  
+# allows testing that node_jasmine_test will work with ts_library from
```
Comment (alexeagle) [inline internal/typescript_mock.bzl]: update comment in this copy-pasta

File: internal/collect_es6_sources.bzl [hunk 6]
```diff
@@ -29,15 +29,24 @@ def collect_es6_sources(ctx):
   """
 
   non_rerooted_files = depset()
+  rerooted_files = []
```
Comment (alexeagle) [inline internal/collect_es6_sources.bzl]: don't move the declaration of this variable up, unless we actually reference it sooner

File: examples/rollup/BUILD.bazel (hunk 1 above)
Comment (mrmeku) [reply examples/rollup/BUILD.bazel]: Donezo

File: examples/rollup/rollup.bzl (hunk 2 above)
Comment (mrmeku) [reply examples/rollup/rollup.bzl]: There's now a cool error message

File: internal/collect_es6_sources.bzl (hunk 3 above)
Comment (mrmeku) [reply internal/collect_es6_sources.bzl]: That already works (I tested it with my angularjs demo is rules_closure)

File: internal/typescript_mock.bzl (hunk 5 above)
Comment (mrmeku) [reply internal/typescript_mock.bzl]: donezo

File: examples/rollup/rollup.bzl [hunk 7]
```diff
@@ -11,17 +11,30 @@
...
+        entry_point = i.path
+        break
+
+  if entry_point == None:
+    fail("Entry point: %s is not valid." % ctx.attr.entry_point)
```
Comment (alexeagle) [inline examples/rollup/rollup.bzl]: can you help them understand what's invalid about it?

File: examples/rollup/rollup.bzl [hunk 8]
```diff
@@ -32,6 +45,7 @@ rollup = rule(
     implementation = _rollup,
     attrs = {
         "srcs": attr.label_list(allow_files = True),
```
Comment (alexeagle) [inline examples/rollup/rollup.bzl]: maybe `allow_files = ['js']`

File: internal/collect_es6_sources.bzl (hunk 3 above)
Comment (alexeagle) [reply internal/collect_es6_sources.bzl]: hmm, I'm not convinced, seems like it should match `file`

File: examples/rollup/rollup.bzl (hunk 8 above)
Comment (mrmeku) [reply examples/rollup/rollup.bzl]: donezo

File: examples/rollup/rollup.bzl (hunk 7 above)
Comment (mrmeku) [reply examples/rollup/rollup.bzl]: donezo

File: internal/typescript_mock.bzl (hunk 4 above)
Comment (mrmeku) [reply internal/typescript_mock.bzl]: donezo

**Comment by alexeagle** [comment]
sorry, I lost track of things, do we still need to merge this?

**Comment by alexeagle** [comment]
@mrmeku sorry this got stale, I'm not clear if we still need to do something here? Do you want to re-base it or should I just close this one?

**Comment by mrmeku** [comment]
Only if you want me to. Otherwise we should just close it

**Comment by alexeagle** [comment]
IIUC this is adding some test coverage? I suppose we get that from downstream projects that use ts_library -> rollup_bundle (like Angular) so we'd catch a bug in pre-release testing.
I'm closing just to clean up old PRs but please keep me honest about merging stuff that we need :)

**Pull request closed without merging**